syntax = "proto3";
import "google/protobuf/empty.proto";
import "task_manager.proto";

option objc_class_prefix = "RTG";


// Interface exported by the server.
service Master {
      rpc request_experiments(ExperimentStatement) returns (MasterResponse) {}

      rpc kill_task(Task) returns (TaskStatus) {}

      rpc get_task_status(Task) returns (TaskStatus) {}

      rpc get_task_log(Task) returns (TaskLog) {}

      rpc get_all_tasks(google.protobuf.Empty) returns (AllTasksStatus) {}

      rpc halt_process_monitor(google.protobuf.Empty) returns (google.protobuf.Empty) {}
}

message ExperimentStatement {
    string name = 1;
    repeated MasterTaskStatement tasks = 2;
}
message MasterTaskStatement {
    string command = 1;
    string name = 2;
    map<string, string> task_env = 4;
}
message MasterResponse {
    string experiment_id = 1;
    enum ResponseStatus {
        SUCCESS = 0;
        FAIL = 1;
    }
    ResponseStatus response = 2; // response가 의미하는 값은 Enum으로 정의
}
